src/js/content/checkWorkerEndpointCSP.ts

```diff
@@ -90,13 +68,61 @@ export function checkWorkerEndpointCSP(
     }
   }
 
-  updateCurrentState(
-    STATES.INVALID,
-    `Worker allows blob:/data: importScripts/import`,
-  );
   return false;
 }
 
+/**
+ * This function should not have side-effects (no throw, no invalidation).
+ * See checkWorkerEndpointCSP for enforcement.
+ */
+export function isWorkerEndpointCSPValid(
+  response: chrome.webRequest.WebResponseCacheDetails,
+  documentWorkerCSPs: Array<Set<string>>,
+  origin: Origin,
+): [true] | [false, string] {
+  const host = ORIGIN_HOST[origin];
+  const cspHeaders = getCSPHeadersFromWebRequestResponse(response)
+    .map(h => h.value)
+    .filter((header): header is string => !!header);
+
+  const cspReportHeaders = getCSPHeadersFromWebRequestResponse(response, true)
+    .map(h => h.value)
+    .filter((header): header is string => !!header);
+
+  if (!checkCSPForEvals(cspHeaders, cspReportHeaders)) {
+    return [false, ''];
```

Comment:
How about "Worker allows evals"?

Reply:
Oh an I think I was missing a state update here... oops

Reply:
Oh, you did not really miss it, checkCSPforEvals currently does the invalidation. I’ll update that in an upcoming PR.